api/src/task/certificates/setup_cert_user_task.py

```diff
@@ -0,0 +1,169 @@
+import logging
+import uuid
+from datetime import datetime
+from enum import StrEnum
+
+import click
+from sqlalchemy import select
+
+import src.adapters.db.flask_db as flask_db
+import src.util.datetime_util as datetime_util
+from src.adapters import db
+from src.constants.lookup_constants import UserType
+from src.db.models import staging
+from src.db.models.user_models import (
+    Agency,
+    AgencyUser,
+    AgencyUserRole,
+    LegacyCertificate,
+    Role,
+    User,
+)
+from src.task.ecs_background_task import ecs_background_task
+from src.task.task import Task
+from src.task.task_blueprint import task_blueprint
+
+logger = logging.getLogger(__name__)
+
+FUTURE_DATE = datetime_util.make_timezone_aware(datetime(2050, 1, 1), "US/Eastern").date()
+
+
+class SetupCertUserTaskStatus(StrEnum):
+    INVALID_ROLE_IDS = "Invalid role ids"
+    AGENCY_NOT_FOUND = "Agency not found"
+    LEGACY_CERTIFICATE_ALREADY_EXISTS = "LegacyCertificate already exists"
+    SUCCESS = "Success"
+    TCERTIFICATE_IS_EXPIRED = "Tcertificate is expired"
+    TCERTIFICATE_NOT_FOUND = "Tcertificate not found"
+    TCERTIFICATE_IS_MISSING_SERIAL_NUMBER = "Tcertificate is missing serial number"
+
+
+@task_blueprint.cli.command("setup-cert-user", help="Setup the LegacyCertificate and User")
+@click.option("--tcertficates-id", "-t", help="tcertificates_id on Staging Tcertificate")
+@click.option("--role-ids", "-t", help="role_id of role that needs to be added", multiple=True)
+@flask_db.with_db_session()
+@ecs_background_task(task_name="setup-cert-user")
+def setup_cert_user(db_session: db.Session, cert_id: str, role_ids: list[str]) -> None:
+    SetupCertUserTask(db_session, cert_id, role_ids).run_task()
+
+
+class SetupCertUserTask(Task):
+
+    def __init__(self, db_session: db.Session, tcertificates_id: str, role_ids: list[str]):
+        super().__init__(db_session)
+        self.tcertificates_id = tcertificates_id
+        self.role_ids = role_ids
+        self.valid_expiration_date = FUTURE_DATE
+        self.top_agency: Agency | None = None
+
+    def run_task(self) -> SetupCertUserTaskStatus:  # type: ignore [override]
```

Comment:
Don't change the type of this function. Do this pattern:

```py
def run_task(self) -> None:
     with db_session.begin():
          self.setup_cert()

def setup_cert(self) -> SetupCertUserTaskStatus:
    ...
```
That also makes it so we don't have to indent all of this.